phpBB/assets/javascript/installer.js

```diff
@@ -564,4 +600,18 @@
 			submitForm($form, $(this));
 		});
 	}
+
+	/**
+	 * Queries the installer's status
+	 */
+	function queryInstallerStatus()
+	{
+		var data = null;
+		$.ajax({url: "../installer/status", dataType: "json"})
+		.done(function(d) {
+			data = d;
+		});
+
+		return data.status;
```

Comment:
This won't work. `data` will be `null` because the request won't yet have returned. Use a callback instead.

Have you tested this? This will throw "Cannot call property status of null".